tests/LoaderTest.php

```diff
@@ -71,4 +75,34 @@ public function testLoader(): void
             $this->assertSame($originalException, $e->getPrevious());
         }
     }
+
+    public function testNotModifiedResponse(): void
+    {
+        $config = new Config();
+        $storage = ComposerFileStorage::create('https://example.net/packages', $config);
+
+        $method = new \ReflectionMethod($storage, 'write');
+        $method->setAccessible(true);
+        $method->invoke($storage, 'test', 'Some test data.');
+        $modifiedTime = $storage->getModifiedTime('test')->format('D, d M Y H:i:s');
+
+        $downloader = $this->prophesize(HttpDownloader::class);
+        $options = [
+            'max_file_size' => 1025,
+            'http' => [
+                'header' => [
+                    "If-Modified-Since: $modifiedTime GMT",
+                ],
+            ],
+        ];
+        $url = '2.test.json';
+        $response = new Response(['url' => $url], 304, [], null);
+        $downloader->get($url, $options)->willReturn($response)->shouldBeCalled();
+
+        $loader = new Loader($downloader->reveal(), $storage);
+        // Since the response has no actual body data, the fact that we get the contents
+        // of the file we wrote here is proof that it was ultimately read from persistent
+        // storage by the loader.
```

Comment:
Should we assert that we don't call getBody or getContent on the response. Basically that we don't read it

Reply:
Sure, I could do that, if I switch to a mock `Response`. Fixed.